src/controllers/controller.bar.js

```diff
@@ -301,21 +307,23 @@ module.exports = DatasetController.extend({
 					imeta.controller._getValueScaleId() === scale.id &&
 					chart.isDatasetVisible(i)) {
 
-					ivalue = +scale.getRightValue(datasets[i].data[index]);
-					if ((value < 0 && ivalue < 0) || (value >= 0 && ivalue > 0)) {
+					yStackValue = scale._parseValue(datasets[i].data[index]);
+					ivalue = yStackValue.start === undefined ? yStackValue.end : yStackValue.min >= 0 && yStackValue.max >= 0 ? yStackValue.max : yStackValue.min;
```

Comment:
Does this work for `[2, -2]` and `[-2, 2]`? I'm not sure how that would be stacked though. 
I think you could use the `size` here if added in `_parseValue`.

Reply:
no at all. The current logic of float-bars in context of stacking means you stack them comparing some base line. By default its 0, so all positive bars will be stacked above and negative - below. Current float bar allows crate bars that crossing 0 line, but if stack such bar it will not be fully correct because of the general stacking algorithm. What you propose can be possible but its another story :)

Reply:
I think this can wait for an actual use case.